src/Microsoft.Dnx.Tooling/NuGet/Core/Configuration/Settings.cs

```diff
@@ -766,15 +767,53 @@ private static Settings ReadSettings(IFileSystem fileSystem, string settingsPath
             }
         }
 
+#if DNXCORE50
+        private static Mutex _globalMutex = new Mutex(initiallyOwned: false);
```

Comment:
This code is from NuGet codebase and DNU actually never runs this code path in multiple threads. To simplify the fix, I just use a global mutex on cross-plat CoreCLR to do the synchronization